server/test/unit/vitalsTest/checkButtonHeartbeatTest.js

```diff
@@ -260,4 +299,151 @@ describe('vitals.js unit tests: checkButtonHeartbeat', () => {
       expect(db.updateButtonsSentVitalsAlerts).to.not.be.called
     })
   })
+
+  describe("when client that is sending vitals's two buttons that are sending vitals last seen exceeds the threshold and no alerts have been sent", () => {
+    beforeEach(async () => {
+      // Set up scenario where one client has two buttons that have just disconnected
+      this.client = factories.clientFactory({ isSendingVitals: true })
+      this.buttonA = buttonFactory({ id: 'fakeIdA', displayName: 'Unit A', sentVitalsAlertAt: null, isSendingVitals: true, client: this.client })
+      this.buttonB = buttonFactory({
+        id: 'fakeIdB',
+        displayName: 'Unit B',
+        buttonSerialNumber: 'AB12-23456',
+        sentVitalsAlertAt: null,
+        isSendingVitals: true,
+        client: this.client,
+      })
+      this.buttonsVitalA = buttonsVitalFactory({
+        createdAt: exceededThresholdTimestamp,
+        button: this.buttonA,
+      })
+      this.buttonsVitalB = buttonsVitalFactory({
+        createdAt: exceededThresholdTimestamp,
+        button: this.buttonB,
+      })
+      sandbox.stub(db, 'getRecentButtonsVitals').returns([this.buttonsVitalA, this.buttonsVitalB])
+      sandbox.stub(db, 'getInactiveGatewaysWithClient').returns([])
+    })
+
+    it('should log three Sentry messages', async () => {
+      await vitals.checkButtonHeartbeat()
+      expect(helpers.logSentry).to.be.calledThrice
+    })
+
+    it('should send the initial disconnection messages for both buttons to Sentry', async () => {
+      await vitals.checkButtonHeartbeat()
+      expect(helpers.logSentry).to.be.calledWith(
+        `Disconnection: ${this.buttonA.client.displayName} ${this.buttonA.displayName} Button delay is ${heartbeatThreshold + 1} seconds.`,
+      )
+      expect(helpers.logSentry).to.be.calledWith(
+        `Disconnection: ${this.buttonB.client.displayName} ${this.buttonB.displayName} Button delay is ${heartbeatThreshold + 1} seconds.`,
```

Comment:
In general, it's better to have just a single `expect` per unit test. Because a test function will stop immediately when an `expect` fails, if the first one fails then it won't even try to run the second, thus hiding a piece of information that might be useful to us when looking at the test results.

Reply:
Good point! I'll split these two up.